patch for fixing #598932, a segfault under certain circumstances due to
wrong node computation if the .info file is malformed.
The patch is based on the one checked in to texinfo cvs by 
Sergey Poznyakoff
---
 info/nodes.c |   13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

Index: texinfo-4.13a.dfsg.1/info/nodes.c
===================================================================
--- texinfo-4.13a.dfsg.1.orig/info/nodes.c	2010-10-08 10:00:22.000000000 +0900
+++ texinfo-4.13a.dfsg.1/info/nodes.c	2010-10-08 10:13:25.000000000 +0900
@@ -960,6 +960,9 @@
         /* If we were able to find this file and load it, then return
            the node within it. */
         {
+          if (!(tag->nodestart >= 0 && tag->nodestart < subfile->filesize))
+            return NULL;
+
           NODE *node = xmalloc (sizeof (NODE));
           node->filename    = subfile->fullpath;
           node->parent      = NULL;
@@ -997,10 +1000,16 @@
 
               min = max = DEFAULT_INFO_FUDGE;
 
-              if (tag->nodestart < DEFAULT_INFO_FUDGE)
+              if (strict_node_location_p)
+                {
+                  min = 0;
+                  max = 2;
+                }
+
+              if (tag->nodestart < min)
                 min = tag->nodestart;
 
-              if (DEFAULT_INFO_FUDGE >
+              if (max >
                   (subfile->filesize - tag->nodestart))
                 max = subfile->filesize - tag->nodestart;
 
